lib/client/connect.js

```diff
@@ -1,19 +1,59 @@
 KadiraDebug = window.KadiraDebug = {};
 KadiraInfo = new Mongo.Collection('kdInfo');
 
-var debug = getQueryParamFromURI("kadira_debug");
-var accessToken = (debug)? (getQueryParamFromURI("access_token") || null) : null;
-Meteor.subscribe('kadira.debug.client.auth', accessToken);
+var env = Utils.getAppEnv();
 
-Meteor.subscribe('kadira.debug.client.listeners');
+var debug = getQueryParamFromURI('kadira_debug');
+var accessToken = (debug)? (getQueryParamFromURI('access_token') || null) : null;
+
+if(env === 'production' && !accessToken) {
+  throw new Meteor.Error('401', 'Unauthorized.');
```

Comment:
No we don't need to throw an error. This is the normal situation. We need to return.